test/test_action.py

```diff
@@ -33,56 +34,98 @@ def test_parse_directory_configs():
 
 def test_no_commit_if_not_dirty():
     """Test no commit created if file is not dirty"""
-    notice_file_path = "./testbench/git/my-notice-file.md"
-
+    repo_root = os.path.join(ROOT_DIR, "testbench/git")
+    notice_file_name = "my-notice-file.md"
+    notice_file_path = os.path.join(repo_root, notice_file_name)
     # init repo
-    repo = Repo.init("./testbench/git")
+    repo = Repo.init(repo_root)
+    subprocess.run(
+        [
+            "git",
+            "config",
+            "--global",
+            "--add",
+            "safe.directory",
+            repo_root,
+        ]
+    )
 
     # create file
     with open(notice_file_path, "w+", encoding="utf-8") as notice_file:
         notice_file.write("first license")
 
     # add file initially
-    repo.index.add("my-notice-file.md")
+    repo.index.add(notice_file_name)
     repo.index.commit("Initial")
 
-    commit_notice_file_if_dirty("./testbench/git/", "my-notice-file")
     ref_log = repo.active_branch.log()
 
     try:
+        assert not is_dirty(repo_root, notice_file_name)
         assert len(ref_log) == 1
         assert ref_log[0].message == "commit (initial): Initial"
     finally:
         os.remove(notice_file_path)
-        shutil.rmtree("./testbench/git/.git")
+        shutil.rmtree(os.path.join(repo_root, ".git"))
```

Comment:
Is "repo_root" and "ROOT_DIR" the same folder?
Looks like bad idea to perform such command in finally statement...

Reply:
no, in the test a new repo is created in /testbench/git, ROOT_DIR is the real repository root

At the end we want to delete the newly created repo again, because it was just used for the test